File: Proyecto LP/mi-servidor/Frontend/lib/Enums/Tipo.dart
enum Tipo {
  sedan("Sedan", "SEDAN"),
  coupe("Coupe", "COUPE"),
  convertible("Convertible", "CONVERTIBLE"),
  hatchback("Hatch-Back", "HATCHBACK"),
  suv("SUV", "SUV"),
  pickUp("Pick-Up", "PICK_UP"),
  hibrido("Hibrido", "HIBRIDO"),
  limosina("Limosina", "LIMOSINA"),
  electrico("Electrico", "ELECTRICO");

  final String displayName;
  final String backendValue;

  const Tipo(this.displayName, this.backendValue);

  static String? getBackendValue(String? displayName) {
    return Tipo.values.firstWhere(
      (tipo) => tipo.displayName == displayName,
      orElse: () => throw Exception("Valor inválido para el enum Tipo: $displayName"),
    ).backendValue;
  }
}

File: Proyecto LP/mi-servidor/Backend/src/main/java/com/miapp/mi_servidor/Enums/Tipo.java
package com.miapp.mi_servidor.Enums;


public enum Tipo {
    SEDAN("Sedan"),
    COUPE("Coupe"),
    CONVERTIBLE("Convertible"),
    HATCHBACK("Hatch-Back"),
    SUV("SUV"),
    PICK_UP("Pick-Up"),
    HIBRIDO("Hibrido"),
    LIMOSINA("Limosina"),
    ELECTRICO("Electrico");

    private final String displayName;

    Tipo(String displayName) {
        this.displayName = displayName;
    }

    public String getDisplayName() {
        return displayName;
    }

    public static Tipo fromValue(String value) {
        for (Tipo tipo : Tipo.values()) {
            if (tipo.getDisplayName().equalsIgnoreCase(value)) {
                return tipo;
            }
        }
        throw new IllegalArgumentException("Valor inválido para el enum Tipo: " + value);
    }
}
